.gitignore

```diff
@@ -65,3 +65,7 @@ __pycache__
 
 # Doxygen outputs
 docs/html
+
+# Shared libs & JAR libs
+*.so
```

Comment:
It may be useful to add a comment on why we generate .so and .jar in source (i.e. that our scripts copy stuff for easy java builds)

Reply:
Narrow this please.

We don't want these files in the source tree, please narrow this so that it only applies an exception to Android app. Probably add a new .gitignore file in the Android directory.

Reply:
Updated, please review.